### ISSUE TITLE
`numpy.arange` returns `Unknown` type with 1.1.395

### ISSUE THREAD

**Issue by tpgillam** [microsoft/pyright#10003]
Consider this code:

```python
import typing

import numpy

typing.reveal_type(numpy.arange(1, 3))
```

Testing with pyright `1.1.394`:
```console
> uv run --with "pyright==1.1.394" pyright moo.py
.../moo/moo.py
  .../moo/moo.py:5:20 - information: Type of "numpy.arange(1, 3)" is "ndarray[tuple[int], dtype[signedinteger[Any]]]"
0 errors, 0 warnings, 1 information
```

and `1.1.395`:
```console
> uv run --with "pyright==1.1.395" pyright moo.py
.../moo/moo.py
  .../moo/moo.py:5:20 - information: Type of "numpy.arange(1, 3)" is "Unknown"
0 errors, 0 warnings, 1 information
```

(Possibly this is related to #9994, but filing separately as I haven't dug into it to be sure)

**Comment by jorenham** [comment]
what version of numpy are you using?

**Comment by erictraut** [comment]
Thanks for the bug report. This turns out to be the same problem as #9984. I'm going to do a hotfix release to address this.

**Comment by erictraut** [comment]
I've published a hotfix release of pyright (1.1.396) to address this issue. I reverted the previous bug fix that caused this regression. The new version is already published to npm and should be available in the VS Code marketplace in the next few minutes and in pypi within the next 24 hours. Apologies for any inconvenience this caused.
